File: Lesson1/scala/com/lesson1/basic/scala/AkkaDB.scala
package com.lesson1.basic.scala

import akka.actor.Actor
import akka.event.Logging
import java.util.HashMap

class AkkaDB extends Actor {
  
  val log = Logging.getLogger(context.system, this)
  
  val map = new HashMap[String, Object]()
  
  override def receive() = {
    case SetRequest(key, value) => {
      log.info("SetRequest")
      log.debug("Key : "+key+" value : "+value)
      map.put(key, value)      
    }
    case _ => log.error("Invalid message : ")
  }
  
}

File: java/Lesson1/src/com/lesson1/basic/AkkaDB.java
package com.lesson1.basic;

import java.util.HashMap;
import java.util.Map;

import akka.actor.AbstractActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.japi.pf.ReceiveBuilder;

public final class AkkaDB extends AbstractActor {

	private final LoggingAdapter LOGGER = Logging.getLogger(context().system(), this);
	private final Map<String, Object> map = new HashMap<>();

	public AkkaDB() {
		receive(ReceiveBuilder.match(SetRequest.class, message -> {
			LOGGER.info("Found key : " + message);
			map.put(message.getKey(), message.getValue());
		})
		.matchAny(message -> LOGGER.info("Unknown Message Found : "+message))

		.build());
	}

	public Map<String, Object> getMap() {
		return map;
	}
	
	

}
